plugins/module_utils/mh/mixins/state.py

```diff
@@ -8,32 +8,37 @@
 
 
 class StateMixin(object):
-    state_param = 'state'
+    state_param = None
     default_state = None
+    default_state_param = 'state'
 
-    def _state(self):
-        state = self.module.params.get(self.state_param)
+    @property
+    def _state_param_name(self):
+        param_name = self.state_param or self.default_state_param
+        if param_name in self.module.params:
```

Comment:
The old code was using the state *value* to do this lookup, while the new one uses the state *option name*. How about simply keeping the old logic and deprecating it, so we can remove it in c.g 5.0.0?

Reply:
@russoz ^

Reply:
Hi @felixfontein yes can do. Down side, though: this is not *a module* code, so we cannot easily throw in a feature flag and give the user a choice on what behaviour to use. That said, do you suggest a strategy to roll out the new code *before* that old one is deprecated?

The only t hing that comes to mind right now is: create another class (say `StateMixinAlias` or `StateMixin50`), then add a warning notice in the original one advising people to use the new one if they are using aliases.

Reply:
The feature flag would be a class property, similar to `state_param` and `default_state`. (The user here is the module author, not the module user.)